lib/diff.ml

```diff
@@ -168,6 +197,73 @@ and type_exprs ~typing_env ~ref_params ~cur_params ~reference ~current =
       type_expr ~typing_env ~ref_params ~cur_params ref cur)
     ~reference ~current
 
+and arrow ~typing_env ~ref_params ~cur_params ~reference ~current =
+  let arg_type lbl typ =
+    match lbl with
+    | Asttypes.Nolabel | Labelled _ -> typ
+    | Optional _ -> (
+        match Types.get_desc typ with
+        | Tconstr (_, [ te ], _) -> te
+        | _ -> assert false)
```

Comment:
This function should be given a proper name and a bit of documentation as to why it's needed!

What do you think of `unwrap_optional_arg`?